contrib/gocql/gocql/option.go

```diff
@@ -18,21 +18,26 @@ import (
 
 const defaultServiceName = "gocql.query"
 
-type queryConfig struct {
-	serviceName, resourceName    string
-	querySpanName, batchSpanName string
-	noDebugStack                 bool
-	analyticsRate                float64
-	errCheck                     func(err error) bool
-	customTags                   map[string]interface{}
-	clusterTagLegacyMode         bool
+type config struct {
+	serviceName, resourceName            string
+	querySpanName, batchSpanName         string
+	noDebugStack                         bool
+	analyticsRate                        float64
+	errCheck                             func(err error) bool
+	customTags                           map[string]interface{}
+	clusterTagLegacyMode                 bool
+	traceQuery, traceBatch, traceConnect bool
 }
 
 // WrapOption represents an option that can be passed to WrapQuery.
-type WrapOption func(*queryConfig)
+type WrapOption func(*config)
 
-func defaultConfig() *queryConfig {
-	cfg := &queryConfig{}
+func defaultConfig() *config {
+	cfg := &config{
+		traceQuery:   true,
```

Comment:
^ somewhere in comments/docs we need to know this is `true` by default, go convention is to have default be `false` so this is a bit surprising. The alternative (a `doNotTraceQuery` field) may even be worse so let's keep those straight, easy to reason about fields, but just add a comment/description somewhere that explains, by default, enabled, traced.

Reply:
I think it would be slightly strange that a package / functions related to tracing `gocqltrace.CreateTracedSession` do not have tracing enabled by default. We already do something similar in some other instrumentation packages that follow a similar pattern (we might disable by default some more low-level / verbose operations the instrumentation might provide).

I've updated the comment in the `With...` options clarifying these are enabled by default.